src/audio/mux/mux.c

```diff
@@ -200,7 +205,229 @@ static struct comp_dev *mux_new(const struct comp_driver *drv,
 	dev->state = COMP_STATE_READY;
 	return dev;
 }
+#else
+static int build_config(struct comp_data *cd, struct comp_dev *dev)
+{
+	dev->ipc_config.type = SOF_COMP_MUX;
+	cd->config.num_streams = 2;
+	int mask = 1;
+	int i, j;
+
+	/* clear masks */
+	for (i = 0; i < cd->config.num_streams; i++)
```

Comment:
Is that verified somewhere that no OOB occurs when iterating over cd->config.streams[] using num_streams as the limit?

Reply:
I do not think so, because the copy function takes into account only those streams/channels on which the mask is set, and this function is performed only for ipc4, where we have only two streams. The change has been tested.

Reply:
Still there is no clear relation between the limit of the loop and the allocation size in this function, code looks unsafe.